adds WindowInfoTest class for mac driver

hkimbrough22 opened the pull request.

dbwiddis reviewed (changes requested):
Looks good.  Your assumption on layer meaning for macOS isn't right, though.

dbwiddis commented on oshi-core/src/test/java/oshi/driver/mac/WindowInfoTest.java:
```diff
@@ -0,0 +1,34 @@
+                assertThat("Window IDs should be unique.", window.getWindowId(), is(not(in(windowIds))));
+                windowIds.add(window.getWindowId());
+                assertThat("Window order number must be greater than or equal to 0", window.getOrder(), is(greaterThanOrEqualTo(0)));
+                assertThat("Window order number must be less than the number of windows", window.getOrder(), is(lessThan(osDesktopWindows.size())));
```
These are allegedly the 20 values, but they don't match what I see, so who knows. ¯\\_(ツ)_/¯

https://github.com/planetbeing/xpwn/blob/ac362d4ffe4d0489a26144a1483ebf3b431da899/idevice/QuicktimeSDK/CIncludes/CGWindowLevel.h#L108-L127

hkimbrough22 commented:
I am unsure if you are able to see what I've pushed up as I did not see a Github PR option, but does list my changes when I look at the repo/brach.

These are my changes, if not: https://github.com/oshi/oshi/commit/11e63179f5ec14b84834b674019cb099f6720e69 

I also wanted to say that I removed the test for screensaver layer as it kept giving me an error on my end since I apparently did not have that layer. I also removed the test for windowIDs being unique as I was getting some errors where they were not unique. Not sure if that is how you're wanting to driver to function, but I removed them in case I was just doing something incorrect!

Please let me know if I can do anything else with this or if I need to do something specific to make a PR (still new to most of this so I'm not sure if it just attaches to my previous PR or not... 😢 ).

dbwiddis commented:
> I am unsure if you are able to see what I've pushed up as I did not see a Github PR option

Pushing to the same branch updates the PR.  Your commit is right above your chat message.
![image](https://user-images.githubusercontent.com/9291703/142074673-799de4fb-af24-4110-b359-e21882f9cd3f.png)

That's curious about the duplicate window IDs. Did you look into whether the duplicates formed a pattern in any way?  I'll have to play around with it to see if I can figure out why.  I guess it's not documented as unique, though.

And it looks like on the CI test there's no desktop layer:
> java.lang.AssertionError: Desktop window layer is not present.

So maybe identifying layers by number is a bad idea.

hkimbrough22 commented:
I did attempt to replicate it, but I did not take any screenshots or anything (probably should have). It was giving me unique windowIDs in the error each time. So one time it would say `67543` has a duplicate, then `8099` has a duplicate. I figured I was just misunderstanding these IDs and they maybe aren't actually unique. 

Should I just remove the layer tests altogether?

dbwiddis commented:
Yeah, I am not sure the server version of macOS even has a desktop configured, so go ahead and remove those tests (the max 20 layers should be ok to leave).  I can't manage to get duplicate window IDs on my machine, I was looking for the process name that might have had them.
